crates/polars-core/src/frame/mod.rs

```diff
@@ -3537,4 +3537,29 @@ mod test {
         assert_eq!(df.get_column_names(), &["a", "b", "c"]);
         Ok(())
     }
+
+    #[cfg(feature = "serde")]
+    #[test]
+    fn test_deserialize_height_validation_8751() {
```

Comment:
Test was moved from Python - the existing test used an exact string representation of the previous JSON format, but that is changed after this PR.